facades/entrypoints.md

```diff
@@ -0,0 +1,185 @@
+## Pre-defined entrypoints
+
+Pre-defined entrypoints inherit from `NetworkEntrypoint` and use sensible default values.
+
+```
+class MainnetEntrypoint extends NetworkEntrypoint(ApiNetworkProvider("https://api.multiversx.com"));
+
+class DevnetEntrypoint extends NetworkEntrypoint(ApiNetworkProvider("https://devnet-api.multiversx.com"));
+
+class TestnetEntrypoint extends NetworkEntrypoint(ApiNetworkProvider("https://testnet-api.multiversx.com"));
+```
+
+## NetworkEntrypoint
+
+The `NetworkEntrypoint` acts as a facade for interacting with the network. It should cover the most common use-cases "out of the box".
+
+**(a)** The facade should allow one to access the underlying, more lower-level components (e.g. transaction controllers, network provider).
+
+**(b)** The facade should promote the most commonly-used functions of the network provider.
+
+**(c)** The facade should fetch network parameters (network config) needed for construction transactions (e.g. chain ID) in a lazy manner. For example, it may fetch them when the first transaction is created. This works fine for JavaScript, as well, since transaction-creation functions are async anyway.
+
+```
+class NetworkEntrypoint:
+    // The constructor is not captured by the specs; it's up to the implementing library to define it.
+    // For example, it can be parametrized with: a network provider URL and kind (proxy, API)
+    constructor(url: str, kind: Optional[str]);
```

Comment:
Is the suggested constructor is all right?

If `kind` is missing, we'd like to infer the kind of network provider based on its URL.